File: ApiServer/Model/views/PermissionModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ApiServer.Model.views
{
    public static class PermissionModel
    {
        public class PermissionBase
        {
            public int Id { get; set; }
            public string Name { get; set; }
            public string Category { get; set; }
            public string DisplayName { get; set; }
            public string Descriptions { get; set; }
        }
        public class PermissionForView : PermissionBase
        {

        }

        public class PermissionWithCategor: PermissionBase
        {
            public bool IsCheck { get; set; }
        }
    }
}

File: Client/src/app/shared/models/permission.model.ts
export interface IPermission {
    id: number
    name: string;
    displayName: string;
    category: string;
    descriptions: string;
}

export class Permission implements IPermission{
    constructor(public id: number, public parent: Permission, public children: Permission[], public name:string, public descriptions: string, public category: string, public displayName: string){}
}

export class PermissionForView implements IPermission{
    constructor(public id: number, public name:string, public descriptions: string, public category: string, public displayName: string){}
}

export class PermissionCategory implements IPermission{
    constructor(public id: number, public name:string, public descriptions: string, public category: string, public displayName: string, public isCheck: boolean){}
}